packages/connectors/ledger-connector/src/iframe/provider.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  IFrameEthereumProvider,
+  IFrameEthereumProviderOptions,
+} from '@ledgerhq/iframe-provider';
+
+interface RequestArguments {
+  readonly method: string;
+  readonly params?: any[];
+}
+
+export class LedgerIFrameProvider extends IFrameEthereumProvider {
+  constructor(options?: IFrameEthereumProviderOptions) {
+    super(options);
+  }
+
+  request({ method, params }: RequestArguments) {
+    return this.sendAsync({ method, params }, (error) => {
+      if (error) console.error(error);
```

Comment:
are you sure you need callback? What if it supresses the the error?

Reply:
replaced with "send" method